deploy/docker/api.py

```
@@ -453,11 +467,45 @@ async def handle_crawl_request(
+        try:
+            results_gen = await partial_func()
+            # Handle both list and async generator results
+            if hasattr(results_gen, '__aiter__'):
+                # It's an async generator, collect all results
+                results = []
+                async for result in results_gen:
+                    results.append(result)
+            else:
+                # It's already a list
+                results = results_gen
+        except Exception as bulk_error:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)